sensor/common/scannerdefinitions/http_handler.go

```diff
@@ -4,45 +4,69 @@ import (
 	"io"
 	"net/http"
 	"net/url"
+	"sync/atomic"
 
 	"github.com/pkg/errors"
 	"github.com/stackrox/rox/pkg/clientconn"
 	"github.com/stackrox/rox/pkg/httputil"
+	"github.com/stackrox/rox/pkg/logging"
 	"github.com/stackrox/rox/pkg/mtls"
 	"github.com/stackrox/rox/pkg/set"
 	"github.com/stackrox/rox/pkg/utils"
+	"github.com/stackrox/rox/sensor/common"
 	"google.golang.org/grpc/codes"
 )
 
 const scannerDefsPath = "/api/extensions/scannerdefinitions"
 
 var (
 	headersToProxy = set.NewFrozenStringSet("If-Modified-Since", "Accept-Encoding")
+	log            = logging.LoggerForModule()
 )
 
-// scannerDefinitionsHandler handles requests to retrieve scanner definitions
+// Handler handles requests to retrieve scanner definitions
 // from Central.
-type scannerDefinitionsHandler struct {
-	centralClient *http.Client
+type Handler struct {
+	centralClient    *http.Client
+	centralReachable atomic.Bool
 }
 
 // NewDefinitionsHandler creates a new scanner definitions handler.
-func NewDefinitionsHandler(centralEndpoint string) (http.Handler, error) {
+func NewDefinitionsHandler(centralEndpoint string) (*Handler, error) {
 	client, err := clientconn.NewHTTPClient(mtls.CentralSubject, centralEndpoint, 0)
 	if err != nil {
 		return nil, errors.Wrap(err, "instantiating central HTTP transport")
 	}
-	return &scannerDefinitionsHandler{
+	return &Handler{
 		centralClient: client,
 	}, nil
 }
 
-func (h *scannerDefinitionsHandler) ServeHTTP(writer http.ResponseWriter, request *http.Request) {
+// Notify reacts to sensor going into online/offline mode.
+func (h *Handler) Notify(e common.SensorComponentEvent) {
+	switch e {
+	case common.SensorComponentEventCentralReachable:
+		h.centralReachable.Store(true)
+	case common.SensorComponentEventOfflineMode:
+		h.centralReachable.Store(false)
+	default:
+		log.Infof("Notified with unknown event: %s", e)
```

Comment:
Let's say we add a third type of event which may not be related to the connectivity to Central. This may not be worth logging. Then again, maybe we'd want to catch any new events that we didn't handle in a `utils.Should(error)` to make sure we account for all events and avoid potential programmer error. What do you think? If we continue logging, maybe this should be a `log.Warnf`

Reply:
Changed to warnf